src/quicklist.h

```diff
@@ -46,8 +48,8 @@
 typedef struct quicklistNode {
     struct quicklistNode *prev;
     struct quicklistNode *next;
-    unsigned char *zl;
-    unsigned int sz;             /* ziplist size in bytes */
+    union {unsigned char *zl; unsigned char *entry;};
+    size_t sz;             /* entry size in bytes */
```

Comment:
can you please elaborate on your suggestion ? not sure I understand.

Reply:
```
unsigned char *zl;
unsigned int type; /* ZIPLSIT=1 or BIGELE=2 */
```
That's what @oranagra should mean.

Reply:
we currently have:
```
    unsigned int extra : 10; /* more bits to steal for future usage */
```
so instead it could be:
```
    unsigned int has_sz_high; /* denotes that we have zs_high as the high 32bits of 'sz' */
    unsigned int extra : 9; /* more bits to steal for future usage */
    int sz_high[0];
```

@yoav-steinberg @yossigo do you think we should do it? i tend to think we should skip it.

Reply:
i.e. then when we allocate "plain" nodes, we allocate them with extra 4 bytes, and look at `(node->sz | (size_t)node->sz_high[0])` rather than plain `node->sz` (obviously we can wrap that in a macro)

Reply:
so what is the bottom line here ? do we want to do it ?

Reply:
i don't want to do that (yet), but let's leave this comment unresolved to get more feedback.

Reply:
The complete bit-field here is:
```c
    unsigned int count : 16;     /* count of items in ziplist */
    unsigned int encoding : 2;   /* RAW==1 or LZF==2 */
    unsigned int container : 2;  /* NONE==1 or ZIPLIST==2 */
    unsigned int recompress : 1; /* was this node previous compressed? */
    unsigned int attempted_compress : 1; /* node can't compress; too small */
    unsigned int extra : 10; /* more bits to steal for future usage */
```
We already have `sz` which is 32 bits. We don't need the `count` field and we don't need the `extra` field. This is another 26 bits. Together with `sz` it's 58 bits (or 67 million times more than 4GB). Why not just use this?

Reply:
why don't we need the `count` field? do you suggest to rely on the ziplist header?
not sure i'd like that.. and also, it may be LZF compressed.

Reply:
I thought `count` is the number of items in the ziplist but in case of NONE container we don't have a zip list but rather just an sds, which is single entry (count always == 1). I guess I'm mixed up.

Reply:
ohh, ok. re-purpose these bits just in case of a PLAIN (non-packed) node.
well, it'll be a little bit complex.. so the question again is if it wort it.
i.e. with all that complexity we might as well just add a few more bytes to the node when it's a PLAIN node and allocate a different struct (the extra bytes are not a high price to pay for a node holding a string that's huge anyway).

but the main question is if the extra 8 bytes do any significant damage on ziplist nodes.
i.e. the extreme case is lots of lists with one small members, but that's not a realistic case, right?

Reply:
I think it's not a big deal to add the extra 8 bytes, but I'm leaning towards the idea of having some union that'll include the "container specific" data (like `extra_len_bits` for NONE and `count` for ZIPLIST). This will make the code clearer I think.
Anyway that's all I have to day, either option is good.